### PR TITLE
Bumped Python SDK version

### LINKED ISSUE #56651: Remove type-check disabling in src/sentry/metrics/minimetrics.py
Once the SDK has been upgraded to one that supports the metrics interface, the `# mypy: ignore-errors` marker has to be removed in the sentry SDK and the non-minimetrics path must be removed.  additionally remove the file from the exclusion in `.pre-commit-config.yaml`

### PR DESCRIPTION
**Pull request by antonpirker**
To the latest and greatest.

Requires https://github.com/getsentry/getsentry/pull/11960

### CODE REVIEW

**Comment by mitsuhiko** [comment]
Something is wrong. The tests should be failing if you move to 1.34.

**Comment by szokeasaurusrex** [comment]
> Something is wrong. The tests should be failing if you move to 1.34.

Why should the tests be failing @mitsuhiko?

**Comment by szokeasaurusrex** [comment]
Commit [8ac1d7e](https://github.com/getsentry/sentry/pull/59123/commits/8ac1d7e182b93eeb7fbf879ff1972d732b0a6dfb) in this PR closes #56651

File: src/sentry/metrics/minimetrics.py [hunk 1]
```diff
@@ -1,27 +1,16 @@
...
 
 import sentry_sdk
-
-try:
-    from sentry_sdk.metrics import Metric, MetricsAggregator, metrics_noop  # type: ignore
-
-    have_minimetrics = True
-except ImportError:
-    have_minimetrics = False
+from sentry_sdk.metrics import Metric, MetricsAggregator, metrics_noop  # type: ignore
```
Comment (mitsuhiko) [inline src/sentry/metrics/minimetrics.py:6]: We should remove the `type: ignore` now.

File: src/sentry/metrics/minimetrics.py [hunk 2]
```diff
@@ -102,8 +91,6 @@ def before_emit_metric(key: str, tags: Dict[str, Any]) -> bool:
 class MiniMetricsMetricsBackend(MetricsBackend):
     def __init__(self, prefix: Optional[str] = None):
         super().__init__(prefix=prefix)
-        if not have_minimetrics:
```
Comment (mitsuhiko) [inline src/sentry/metrics/minimetrics.py:105]: I think the entire constructor can go now.

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

File: src/sentry/metrics/minimetrics.py (hunk 2 above)
Comment (szokeasaurusrex) [reply src/sentry/metrics/minimetrics.py:105]: Good point

**Review by mitsuhiko — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -53,12 +53,6 @@ repos:
       language: python
       files: requirements-.*\.txt$
       additional_dependencies: [packaging==21.3]
-    - id: check-mypy-bypass
-      name: do not bypass the type checker
-      entry: '(^# *mypy: *ignore-errors|^# *type: *ignore|\bno_type_check\b)'
-      language: pygrep
-      types: [python]
-      exclude: ^src/sentry/metrics/minimetrics.py$
```
Comment (asottile-sentry) [inline .pre-commit-config.yaml:61]: only this line should be removed.  the check is very important

**Review by asottile-sentry — DISMISSED** [review]

File: .pre-commit-config.yaml (hunk 3 above)
Comment (szokeasaurusrex) [reply .pre-commit-config.yaml:61]: Good catch, thanks for pointing this out!

**Review by ghost — APPROVED** [review]

**Pull request closed without merging**